src/BuildScriptGenerator/DotNetCore/VersionProviders/DotNetCoreSdkStorageVersionProvider.cs

```diff
@@ -0,0 +1,80 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Xml.Linq;
+using System.Xml.XPath;
+
+namespace Microsoft.Oryx.BuildScriptGenerator.DotNetCore
+{
+    public class DotNetCoreSdkStorageVersionProvider : SdkStorageVersionProviderBase, IDotNetCoreVersionProvider
+    {
+        private Dictionary<string, string> _versionMap;
+        private string _defaultRuntimeVersion;
+
+        public DotNetCoreSdkStorageVersionProvider(IEnvironment environment, IHttpClientFactory httpClientFactory)
+            : base(environment, httpClientFactory)
+        {
+        }
+
+        public Dictionary<string, string> SupportedVersionsMap { get; }
+
+        public string GetDefaultRuntimeVersion()
+        {
+            GetVersionInfo();
+            return _defaultRuntimeVersion;
+        }
+
+        public Dictionary<string, string> GetSupportedVersions()
+        {
+            GetVersionInfo();
+            return _versionMap;
+        }
+
+        public void GetVersionInfo()
+        {
+            if (_versionMap == null)
+            {
+                var httpClient = _httpClientFactory.CreateClient("general");
+                var sdkStorageBaseUrl = GetPlatformBinariesStorageBaseUrl();
+                var blobList = httpClient
+                    .GetStringAsync($"{sdkStorageBaseUrl}/dotnet?restype=container&comp=list&include=metadata")
+                    .Result;
+
+                var xdoc = XDocument.Parse(blobList);
```

Comment:
nit: inside a try? what if a parsing error happens? do we want to log it to telemetry as well?